internal/core/command_interceptor.go

```diff
@@ -0,0 +1,32 @@
+package core
+
+import (
+	"context"
+)
+
+func CombineCommandInterceptor(interceptors ...CommandInterceptor) CommandInterceptor {
```

Comment:
Do we need this command to be public? Shouldn't we rely on the `AddInterceptors` method only?